lib/ui/platform_dispatcher.dart

```diff
@@ -1251,6 +1296,106 @@ class WindowPadding {
   }
 }
 
+/// Area of the display that is obstructed by a hardware feature.
+///
+/// The bounds, which the application can use to guide layout, may be obscured
+/// or not have touch capabilities. The type can be used to determine behaviour.
+/// For example, a hinge feature can be used to separate the layout into 2
+/// logical areas or panels in the application.
+class DisplayFeature {
+  const DisplayFeature({
+    required this.rect,
+    required this.type,
+    required this.state,
+  });
+
+  DisplayFeature withDevicePixelRatio(double devicePixelRatio) {
+    return DisplayFeature(
+      rect : Rect.fromLTRB(
+        rect.left / devicePixelRatio,
+        rect.top / devicePixelRatio,
+        rect.right / devicePixelRatio,
+        rect.bottom / devicePixelRatio,
+      ),
+      type: type,
+      state: state,
+    );
+  }
+
+  /// Area of the view occupied by this display feature
+  final Rect rect;
+
+  /// Type of display feature, e.g. hinge, fold, cutout
+  final DisplayFeatureType type;
+
+  /// Posture of display feature, which is populated only for folds and hinges.
+  /// For cutouts, this is [DisplayFeatureState.unknown]
+  final DisplayFeatureState state;
+
+  @override
+  bool operator ==(Object other) =>
```

Comment:
style nit: we only allow the `=>` operator in specific cases: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#consider-using--for-short-functions-and-methods